### ISSUE TITLE
Fix S3956 FP: it's triggered on sealed classes

### ISSUE THREAD

**Issue by AndreaVielCGN** [SonarSource/sonar-dotnet#9752]
### Description

Rule S3956 is '"Generic.List" instances should not be part of public APIs' or 'Refactor this method to use a generic collection designed for inheritance.' (depending on the place).

The description reports that code shouldn't use List<T> instead of IList<T> because it is a generic collection that is designed for performance and not inheritance. However, the diagnostic is triggered even on **sealed** types, where inheritance is not a concern. 

### Reproducer

```csharp
public sealed class Context {
     public List<string> Problems { get; set; }
}
```

### Product and Version

SonarQube Server v25.2.0.102705 and SonarQube for Visual Studio 8.23.0.13726

**Comment by jilles-sg** [comment]
The note about inheritance is about the `List` itself and not about the containing type. The idea is that a class with behaviour (not merely a DTO) is likely to want to modify the collection's behaviour (for example by adding additional checks, adding methods for special operations or notifying other parts of the class of changes). This concern applies whether or not the class is sealed, since even a sealed class can change in the future.

If a method returns a snapshot of some collection or if the containing type is a mere DTO, `List` or even an array or immutable array would be fine.

Also, rules like these are less useful if binary compatibility is unimportant (all internal source code is recompiled each time).

**Comment by AndreaVielCGN** [comment]
Thank you, I understand your point and I agree. I thought it meant something different. I'll close this.
